base/loading.jl

```diff
@@ -276,6 +276,9 @@ end
 # require always works in Main scope and loads files from node 1
 const toplevel_load = Ref(true)
 
+myid() = isdefined(Main, :Distributed) ? invokelatest(Main.Distributed.myid) : 1
```

Comment:
I am worried about the performance overhead of `invokelatest` here.
It might be faster to define `myid() = 1` here and upon loading `Distributed` do a `@eval Base myid() = Distributed.myid()`. 
or we could define: 

```
@eval Base begin
const _Distributed = Base.root_module(:Distributed)
myid() = isdefined(Main, :Distributed) ? _Distributed.myid() : 1
end
```

at the end of `sysimage.jl`
cc @vtjnash

Reply:
Have left this as is for now. The first option results in a build time warning followed by failures in the `compile` test. The second one runs into dependency issues with `require` in sysimg.jl

I'll open an issue to optimize these lines once this PR is merged.